Notebook/Reactjs/learning-hooks/src/useWindowDimensions.js

```diff
@@ -1,31 +1,28 @@
 import { useState, useEffect } from "react";
 
+function debounce(callback, wait) {
+  let timeout;
+  return (...args) => {
+    const context = this;
+    clearTimeout(timeout);
+    timeout = setTimeout(() => callback.apply(context, args), wait);
+  };
+}
+
 const useWindowDimensions = () => {
   const [innerWidth, setInnerWidth] = useState(window.innerHeight);
   const [innerHeight, setInnerHeight] = useState(window.innerWidth);
 
-  const calcWindowSize = () => {
-    const height = window.innerHeight;
-    const width = window.innerWidth;
-    setInnerWidth(width);
-    setInnerHeight(height);
-  };
-  function debounce(callback, wait) {
-    let timeout;
-    return (...args) => {
-      const context = this;
-      clearTimeout(timeout);
-      timeout = setTimeout(() => callback.apply(context, args), wait);
-    };
-  }
-
   useEffect(() => {
-    window.addEventListener(
-      "resize",
-      debounce(() => {
-        calcWindowSize();
-      }, 500)
```

Comment:
```
      debounce(() => {	      const width = window.innerWidth;
        calcWindowSize();	      setInnerWidth(width);
      }, 500)
```

and 

```
calcWindowSize
```

are two different functions, meaning `removeEventListener` won't register the handler.